cloud/services/publicips/publicips.go

```diff
@@ -76,16 +97,46 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	if !ok {
 		return errors.New("invalid PublicIP Specification")
 	}
+
+	ip, err := s.Client.Get(ctx, s.Scope.ResourceGroup(), publicIPSpec.Name)
+	if err != nil {
+		if ip.IsHTTPStatus(http.StatusNotFound) {
+			return errors.Wrapf(err, "failed to find public ip '%s' for deletion, ignoring and finishing successfully", publicIPSpec.Name)
+		}
+		return errors.Wrapf(err, "failed to fetch public ip '%s' to check if we should delete", publicIPSpec.Name)
+	}
+
+	tags := converters.MapToTags(ip.Tags)
+	isManaged := tags.HasOwned(s.Scope.Name())
+	if isManaged {
+		klog.V(2).Infof("found unmanaged public ip %s, will not delete", publicIPSpec.Name)
+		return nil
+	}
+
 	klog.V(2).Infof("deleting public ip %s", publicIPSpec.Name)
-	err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), publicIPSpec.Name)
-	if err != nil && azure.ResourceNotFound(err) {
+	if err := s.Client.Delete(
+		ctx,
+		s.Scope.ResourceGroup(),
+		publicIPSpec.Name,
+	); err != nil && azure.ResourceNotFound(err) {
 		// already deleted
 		return nil
 	}
+
 	if err != nil {
 		return errors.Wrapf(err, "failed to delete public ip %s in resource group %s", publicIPSpec.Name, s.Scope.ResourceGroup())
 	}
 
 	klog.V(2).Infof("deleted public ip %s", publicIPSpec.Name)
 	return err
 }
+
+func (s *Service) isManaged(ctx context.Context, name string) (bool, error) {
```

Comment:
cc @cpanato

This is exactly the change you need for #708

Reply:
will review and follow the approach! thanks!